### PR TITLE
Tests whether mCurrentEditedCard is null

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Tests throughout the whole code, even if the only problem known to date is in:
```
if (!mEditorNote.cids().contains(mCurrentEditedCard.getId())) {
```

This should solve #6719

Testing it on my merge branch and trying to reproduce the bug

### CODE REVIEW

**Comment by david-allison-1** [comment]
I think the main question to ask (likely for a future PR) is whether the variable needs to be static in the first place.

It's persisted from a previous instance of the screen, and doesn't seem like the correct action to take.

**Comment by Arthur-Milchior** [comment]
I added a second commit which simply but this back to null unless it's an edition

**Comment by mikehardy** [comment]
Oh interesting - I hadn't noticed it was static in violation of naming standards (should be sCurrentEditedCard in that case). I wonder who is using it statically? Very wierd. I'm logging a related issue on it - this is the fix for now but this variable smells bad and needs a look

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 1]
```diff
@@ -422,6 +423,7 @@ protected void onCollectionLoaded(Collection col) {
 
 
         mAedictIntent = false;
+        mCurrentEditedCard = null;
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:426]: This seems like the right thing to do as I read the whole onCollectionLoaded, but NoteEditor is a mess so it is hard to say for sure. It's clear from the crash not detected in unit tests that our coverage here isn't great but NoteEditor is scheduled for some serious work soon anyway, so hopefully it gets covered ...

**Pull request merged**